File: 03_Data_Structures_&_Counting/10816_Frequency_Table/Answer10816.kt
fun main(){
  readln()
  val firstList = readln().split(" ")
  readln()
  val queryList = readln().split(" ")
  val frequencyTable = HashMap<String,Int>()
  for(it in firstList) {
    frequencyTable[it] = (frequencyTable[it] ?: 0) + 1
  }
  val resultArr = queryList.map{frequencyTable[it] ?: 0}
  print(resultArr.joinToString(" "))
}

File: 03_Data_Structures_&_Counting/10816_Frequency_Table/Answer10816.ts
const [_, firstList, __, queryList] =`${require('fs').readFileSync(0)}`.split('\n')
    .map(l => l.split(' '))
const frequencyTable = {}
for(const it of firstList) {
  frequencyTable[it] = (frequencyTable[it] ?? 0) + 1
}
const resultArr = queryList.map(it=> frequencyTable[it] ?? 0)
console.log(resultArr.join(' '))